tests/tensorboard_test.py

```diff
@@ -0,0 +1,204 @@
+    # assert the image was increased in dimension
+    self.assertEqual(expected_img.shape, (30, 30, 3))
+
+  def test_summarywriter_single_channel_image_scaled(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = tf.random.uniform(shape=[30, 30, 1])
+    img = tf.cast(255 * img, tf.uint8)
+    summary_writer.image(tag='2dimage_1channel_test', image=img, step=1)
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+
+    self.assertEqual(summary_value.tag, '2dimage_1channel_test')
+    expected_img = tf.image.decode_image(summary_value.tensor.string_val[2])
+    # assert the image was increased in dimension
+    self.assertEqual(expected_img.shape, (30, 30, 3))
+
+  def test_summarywriter_audio(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    audio = tf.random.uniform(shape=[2, 48000, 2], minval=-1.0, maxval=1.0)
+    summary_writer.audio(tag='audio_test', audiodata=audio, step=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'audio_test')
+
+    # Assert two audio files are parsed.
+    self.assertLen(summary_value.tensor.string_val, 4)
+
+    # Assert values.
+    expected_audio = \
+        tf.audio.decode_wav(summary_value.tensor.string_val[0]).audio
+    self.assert_encoded_audio_files_equal(
+        actual_audio=audio[0], expected_audio=expected_audio)
+
+    expected_audio = \
+        tf.audio.decode_wav(summary_value.tensor.string_val[2]).audio
+    self.assert_encoded_audio_files_equal(
+        actual_audio=audio[1], expected_audio=expected_audio)
+
+  def test_summarywriter_audio_sampled_output(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    audio = tf.random.uniform(shape=[2, 48000, 2], minval=-1.0, maxval=1.0)
+    summary_writer.audio(
+        tag='audio_test', audiodata=audio, step=1, max_outputs=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'audio_test')
+
+    # Assert only the first audio clip is available.
+    self.assertLen(summary_value.tensor.string_val, 2)
+
+    # Assert values.
+    expected_audio = \
+        tf.audio.decode_wav(summary_value.tensor.string_val[0]).audio
+    self.assert_encoded_audio_files_equal(
+        actual_audio=audio[0], expected_audio=expected_audio)
+
+  def test_summarywriter_histogram(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    histogram = onp.arange(1000)
+    summary_writer.histogram(tag='histogram_test', values=histogram, step=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'histogram_test')
+    expected_histogram = tensor_util.make_ndarray(summary_value.tensor)
+    self.assertTrue(expected_histogram.shape, (30, 3))
```

Comment:
I'm puzzled by the shape. How does one arrive to (30, 3) ? And what are the entries of `expected_histogram` ?

Reply:
Yeah my bad, the default value of bins assumed by the tf.summary.histogram is 30. Added the comment and adjusted the test name accordingly.

Reply:
Thank you for the clarification!